cmd/trace-agent/config/config.go

```diff
@@ -597,3 +594,44 @@ func acquireHostnameFallback(c *config.AgentConfig) error {
 	log.Debugf("Acquired hostname from core agent (%s): %q.", c.DDAgentBin, c.Hostname)
 	return nil
 }
+
+// SetHandler returns handler for runtime configuration changes.
+func SetHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method != http.MethodPost {
+			http.NotFoundHandler()
+			return
+		}
+		for set, values := range req.URL.Query() {
+			value := ""
+			for i := range values {
+				if v := html.UnescapeString(values[i]); v != "" {
+					value = v
+				}
+			}
+			if value == "" {
+				continue
+			}
+			switch set {
+			case "log_level":
+				lvl := strings.ToLower(value)
+				if lvl == "warning" {
+					lvl = "warn"
+				}
+				if err := coreconfig.ChangeLogLevel(lvl); err != nil {
+					httpError(w, http.StatusInternalServerError, err)
+					return
+				}
+				coreconfig.Datadog.Set("log_level", lvl)
+				log.Infof("Switched log level to %s", lvl)
+			default:
+				log.Infof(fmt.Sprintf("unrecognized setting: %s", set))
```

Comment:
```suggestion
				log.Infof("Unsupported config change requested (key: %q).", set)
```
Let's be specific. It may be a recognised setting, but we don't support changing it yet. Also, `%q` is nicer in log outputs, and always use capital letter).